### DISCUSSION TITLE
Inline Code Highlighting

### DISCUSSION THREAD

**Discussion by babakfp** [remarkjs/remark#1151] [Q&A] [answered]
Hi

I wrote this remark plugin for adding an inline code highlighting feature: https://github.com/babakfp/remark-inline-code-language

But, it seems like I should've made the plugin for rehype because I don't get back the `lang` property in this example:

```js
import { unified } from "unified"
import { visit } from "unist-util-visit"
import remarkParse from "remark-parse"
import remarkInlineCodeLanguage from "../../remark-inline-code-language/dist/index.js"
import remarkRehype from "remark-rehype"
import rehypeStringify from "rehype-stringify"

const file = await unified()
	.use(remarkParse)
	.use(remarkInlineCodeLanguage)
	.use(remarkRehype)
	.use(() => {
		return (tree, file) => {
				visit(tree, "element", function visitor(node) {
				console.log(JSON.stringify(node, null, 4))
			})
		}
	})
	.use(rehypeStringify)
	.process("`_js console.log()`")
```

What can I do so rehype adds the `lang` property? I'm looking for an easy solution, otherwise, I'm going to write the same plugin for rehype too :|

Thanks

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @babakfp! 👋 
It looks like you are trying to add a `lang` attribute to an `inlineCode` node?
https://github.com/babakfp/remark-inline-code-language/blob/927f2c41df5b8738ee02f240273ef02de9bf47c5/src/index.ts#L22
https://github.com/babakfp/remark-inline-code-language/blob/927f2c41df5b8738ee02f240273ef02de9bf47c5/src/index.ts#L33

in `mdast` an `inlineCode` node does not have a `lang` attribute https://github.com/syntax-tree/mdast#inlinecode

a few options:
1. turn the node into type `code` instead of `inlineCode`
2. or, if you specifically want to set an attribute which will appear in HTML, use the `.data.hProperties` attribute on the node https://github.com/syntax-tree/mdast-util-to-hast#hproperties

Taking a few steps back, if you want to edit the HTML, this could alternatively be done as a `rehype` plugin, which edits `code` tags.

**Reply by wooorm** [reply]
I’d recommend going with rehype. You care about HTML. Not about markdown.
